libstuff/libstuff.h

```diff
@@ -111,10 +111,15 @@ class SString : public string {
 typedef map<string, SString, STableComp> STable;
 
 // An SException is an exception class that can represent an HTTP-like response, with a method line, headers, and a
-// body. The STHROW and STHROW_STACK macros will create an SException that logs it's file and line of creation, and
-// optionally, a stack trace at the same time. They can take, 1, 2, or all 3 of the components of an HTTP response
-// as arguments.
-#define STHROW(...) throw SException(__FILE__, __LINE__, false, __VA_ARGS__)
+// body. The STHROW and STHROW_STACK macros will create an SException that logs it's file, line of creation, and
+// a stack trace at the same time. They can take, 1, 2, or all 3 of the components of an HTTP response as arguments.
+#define STHROW(...)                                                     \
+do {                                                                    \
+    SException ex = SException(__FILE__, __LINE__, false, __VA_ARGS__); \
+    SLogStackTrace(LOG_DEBUG);                                          \
+    throw ex;                                                           \
```

Comment:
Why not just add the stack log before the existing throw statement instead of splitting the constructor call and the throw?